### PR TITLE
Adding label for sidecar injection in kyma-namespace

### PR DESCRIPTION
**Pull request by dariusztutaj**
- Added labels for sidecar-injection for Kyma-system namespace
- Changed Proxy Reset logic to reset all pods which belong to mesh based on:
  -  enableNamespacesByDefault setting in values.yaml
  - namespace label
  - pod annotations
- updated mocks
- updated and adjusted tests

**Related Issues:**
#1126
https://github.com/kyma-project/kyma/issues/14911

### CODE REVIEW

**Comment by barchw** [comment]
Please regenerate mocks

File: pkg/reconciler/instances/istio/reset/data/gatherer.go [hunk 1]
```diff
@@ -81,26 +83,33 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
...
 	return
 }
 
-func getPodsWithAnnotation(inputPodsList v1.PodList) (outputPodsList v1.PodList) {
+func getPodsWithAnnotation(inputPodsList v1.PodList, sidecarInjectionEnabledbyDefault bool) (outputPodsList v1.PodList) {
 	inputPodsList.DeepCopyInto(&outputPodsList)
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
 		if pod.Annotations["sidecar.istio.io/inject"] == "false" || pod.Annotations["reconciler/namespace-istio-injection"] == "disabled" {
```
Comment (cnvergence) [inline pkg/reconciler/instances/istio/reset/data/gatherer.go:103]:
```suggestion
		if !sidecarInjectionEnabledbyDefault {
				_, namespaceLabeled := pod.Annotations["reconciler/namespace-istio-injection"]
				_, podAnnotated := pod.Annotations["sidecar.istio.io/inject"]
				if !namespaceLabeled && !podAnnotated {
					continue
				}     
		if pod.Annotations["sidecar.istio.io/inject"] == "false" || pod.Annotations["reconciler/namespace-istio-injection"] == "disabled" {
			continue
			}
```

Comment (cnvergence) [reply pkg/reconciler/instances/istio/reset/data/gatherer.go:103]: I think the logic you have added is what we need. My suggestion is to move it out of if-else and check it before checking disabled annotations.

**Comment by cnvergence** [comment]
Please fix the Gatherer mocks and let's test this logic in https://github.com/kyma-incubator/reconciler/blob/main/pkg/reconciler/instances/istio/reset/data/gatherer_test.go#L103

File: pkg/reconciler/instances/istio/reset/data/gatherer_test.go [hunk 2]
```diff
@@ -220,7 +222,162 @@ func Test_Gatherer_GetPodsWithoutSidecar(t *testing.T) {
...
+		gatherer := DefaultGatherer{}
+
+		// when
+		podsWithoutSidecar, err := gatherer.GetPodsWithoutSidecar(kubeClient, retryOpts, sidecarInjectionEnabledByDefault)
+
+		// then
+		require.NoError(t, err)
+		require.Equal(t, 1, len(podsWithoutSidecar.Items))
+	})
+	t.Run("should not get pod without Istio sidecar and annotated sidecar.istio.io/inject=true with in namespace labeled istio-injection=disabled", func(t *testing.T) {
```
Comment (cnvergence) [inline pkg/reconciler/instances/istio/reset/data/gatherer_test.go:338]:
```suggestion
	t.Run("should not get pod without Istio sidecar and annotated sidecar.istio.io/inject=true with in namespace labeled istio-injection=enabled", func(t *testing.T) {
```

File: pkg/reconciler/instances/istio/reset/data/gatherer.go [hunk 3]
```diff
@@ -81,30 +83,39 @@ func (i *DefaultGatherer) GetPodsWithDifferentImage(inputPodsList v1.PodList, im
...
+func getPodsWithAnnotation(inputPodsList v1.PodList, sidecarInjectionEnabledbyDefault bool) (outputPodsList v1.PodList) {
 	inputPodsList.DeepCopyInto(&outputPodsList)
 	outputPodsList.Items = []v1.Pod{}
 
 	for _, pod := range inputPodsList.Items {
-		if pod.Annotations["sidecar.istio.io/inject"] == "false" || pod.Annotations["reconciler/namespace-istio-injection"] == "disabled" {
+		namespaceLabelValue, namespaceLabeled := pod.Annotations["reconciler/namespace-istio-injection"]
+		podAnnotationValue, podAnnotated := pod.Annotations["sidecar.istio.io/inject"]
+
+		if (namespaceLabeled) && (namespaceLabelValue == "disabled") {
```
Comment (cnvergence) [inline pkg/reconciler/instances/istio/reset/data/gatherer.go:106]: nit: do we need parentheses here?

**Review by cnvergence — APPROVED** [review]
LGTM

**Review by tobiscr — APPROVED** [review]

**Pull request merged**